packages/block-editor/src/components/block-list-appender/index.js

```diff
@@ -92,6 +88,26 @@ function BlockListAppender( {
 	renderAppender,
 	className,
 	tagName: TagName = 'div',
+} ) {
+	if ( renderAppender === false ) {
+		return null;
+	}
```

Comment:
What we're mostly doing is moving this condition a level above and moving the rest to a subcomponent. That way if the appender is `false`, we'll not have to create extra subscriptions because the subcomponent will never be rendered.